src/main/java/org/mockito/internal/junit/ArgMismatchFinder.java

```diff
@@ -0,0 +1,28 @@
+package org.mockito.internal.junit;
+
+import org.mockito.internal.invocation.finder.AllInvocationsFinder;
+import org.mockito.invocation.Invocation;
+import org.mockito.internal.invocation.Stubbing;
+
+/**
+ * For given mocks, finds stubbing arg mismatches
+ */
+class ArgMismatchFinder {
+
+    StubbingArgMismatches getStubbingArgMismatches(Iterable<Object> mocks) {
```

Comment:
I like it seperate. Cleaner OO, separation of complexities into objects/classes. easy testing.